hangman/src/index.js

```diff
@@ -0,0 +1,288 @@
+mainContainerPart1.appendChild(gallowsContainer);
+mainContainerPart1.appendChild(title);
+
+// Create game div
+const game = document.createElement('div');
+game.id = 'game';
+
+// Create word to guess paragraph
+const wordToGuess = document.createElement('p');
+wordToGuess.id = 'wordToGuess';
+wordToGuess.innerHTML = '<span id="word"></span>';
+
+// Create hint paragraph
+const hint = document.createElement('p');
+hint.id = 'wordToGuess';
+hint.innerHTML = '<span id="hint"></span>';
+
+// Create guesses paragraph
+const guesses = document.createElement('p');
+guesses.id = 'guesses';
+guesses.innerHTML = '<span id="wrongGuesses"></span>';
+
+// Create status paragraph
+const statusDisplay = document.createElement('p');
+statusDisplay.id = 'status';
+
+// Create keyboard div
+const keyboard = document.createElement('div');
+keyboard.id = 'keyboard';
+
+// Append game elements to the game div
+game.appendChild(wordToGuess);
+game.appendChild(hint);
+game.appendChild(guesses);
+game.appendChild(statusDisplay);
+game.appendChild(keyboard);
+
+// Append game to the main container part
+const mainContainerPart2 = document.createElement('div');
+mainContainerPart2.classList.add('main-container__part');
+mainContainerPart2.appendChild(game);
+
+// Append both parts to the main container
+mainContainer.appendChild(mainContainerPart1);
+mainContainer.appendChild(mainContainerPart2);
+
+// Append the main container to the body
+document.body.appendChild(mainContainer);
+
+const gallows = document.querySelector('.gallows-container');
+
+function displayPartOfBody(partsDrawn) {
+  const bodyPart = document.createElement('img');
+  bodyPart.classList.add('bodypart');
+  if (partsDrawn === 1) {
+    bodyPart.src = humanHead;
+    gallows.appendChild(bodyPart);
+  } else if (partsDrawn === 2) {
+    bodyPart.src = humanBody;
+    gallows.appendChild(bodyPart);
+  } else if (partsDrawn === 3) {
+    bodyPart.src = humanArmLeft;
+    gallows.appendChild(bodyPart);
+  } else if (partsDrawn === 4) {
+    bodyPart.src = humanArmRight;
+    gallows.appendChild(bodyPart);
+  } else if (partsDrawn === 5) {
+    bodyPart.src = humanLegLeft;
+    gallows.appendChild(bodyPart);
+  } else if (partsDrawn === 6) {
+    bodyPart.src = humanLegRight;
+    gallows.appendChild(bodyPart);
+  }
+}
+
+const url = './src/questions.json';
+let jsonData;
+try {
+  let response = await fetch(url);
+
+  if (response.ok) {
+    // если HTTP-статус в диапазоне 200-299, получаем тело ответа
+    jsonData = await response.json();
+  } else {
+    alert('Error: ' + response.status);
+  }
+} catch (error) {
+  // обработка ошибок при запросе
+  alert('Error: ' + error);
+}
+
+/* Game Constants */
+
+const maxWrongGuesses = 6;
+
+const wordDisplay = document.querySelector('#word');
+const hintDisplay = document.querySelector('#hint');
+const wrongGuessesDisplay = document.querySelector('#wrongGuesses');
+
+/* Game State */
+
+let word = pickRandom(jsonData);
+let rightGuesses = [];
```

Comment:
I would use `Set` for `rightGuesses` <https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Set>

Faster to check if you have element in Set, impossible to add duplicated letters.